### PR TITLE
Feature mark conflicting executions

### LINKED ISSUE #1696: Parallel processing support
- Smart Contract
   - [X] Metadata definition (1 day)
   - Metadata identification
      - ~~Metadata state path sets (3 days - simplest version, longer for more elegant design)~~
      - [X] Developers to provide the state path sets (we need a service to read it) #1725 
- ~~Metadata storage (maintain graph) (n days)~~
- [X] Service to provide state paths for txn (2 days)
- Execution
   - [x] Find resources (1 day)
   - [x] Group transactions (1 day)
   - [x] Parallel execution
   - Confirm state accesses after execution, reject result if not correctly identified
      - [x] Add read in TransactionExecutingStateSet (1 day)
      - ~~Check read/write match identified state access (1 day)~~
      - ~~Implement group state merging (1 day)~~
      - [x] Label contract as non-parallelizable if merging failed (2 days)

**The estimate days only include feature code. With tests, the time will be longer.*

**Possible Challenges:**
* For 2 stage execution (1 group->n group), i.e. 1st stage no parallel execution, 2nd stage parallel
  Currently all state changes in a block are stored in blockstate, need to pass change in 1st stage to 2nd stage. The traffic may be huge if n is big.

### LINKED ISSUE #1818: Remove. Temporarily added to increase execution time
https://github.com/AElfProject/AElf/blob/2c9c151b530dd6cd04fbbcc9a6bbd4d77ac3ccc1/contract/AElf.Contracts.Consensus.AEDPoS/AEDPoSContract_HelpMethods_GetConsensusCommand.cs#L229-L234

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 2c9c151b530dd6cd04fbbcc9a6bbd4d77ac3ccc1 when #1747 was merged. cc @AElfProject.

### LINKED ISSUE #1821: Add event handler somewhere, identify the conflicting transactions and remove th...
https://github.com/AElfProject/AElf/blob/2c9c151b530dd6cd04fbbcc9a6bbd4d77ac3ccc1/src/AElf.Kernel.SmartContract.Parallel/LocalParallelTransactionExecutingService.cs#L83-L88

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in 2c9c151b530dd6cd04fbbcc9a6bbd4d77ac3ccc1 when #1747 was merged. cc @AElfProject.

### PR DESCRIPTION
**Pull request by gldeng**
close #1818 close #1696
fix #1821

### CODE REVIEW

**Comment by jason-hoopox** [comment]
close #1818, close #1821

File: contract/AElf.Contracts.MultiToken/TokenContract_ACS2_StatePathsProvider.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public override ResourceInfo GetResourceInfo(Transaction txn)
             }
         }
 
-        private int GetPathHashCode(params string[] parts)
+        private ScopedStatePath GetPath(params string[] parts)
```
Comment (loning) [inline contract/AElf.Contracts.MultiToken/TokenContract_ACS2_StatePathsProvider.cs]: why do we use string path instead of int hash value？

Comment (jason-hoopox) [reply contract/AElf.Contracts.MultiToken/TokenContract_ACS2_StatePathsProvider.cs]: We can use string path to get resource keys.However hash value can't.

Comment (loning) [reply contract/AElf.Contracts.MultiToken/TokenContract_ACS2_StatePathsProvider.cs]: Why do we need to get the resource keys? we just need to know, same resource cannot running in parallel, int value compares other int value much faster than compare string or bytes array.

can you make a performance benchmark compare when you finish a big changes, or make the performance test in the CI? @rosona @jason-hoopox

**Pull request merged**